### PR TITLE
fix: copy scope when deriving an atom

### LINKED ISSUE #249: useUpdateAtom scope
After @JLarky created this [sandbox](https://codesandbox.io/s/admiring-pascal-hsnuk?file=/src/library/jotaiProvider.tsx)(L10,jotaiProvider) I found out that useUpdateAtom doesn't set the original atom scope on the new derived-writable atom. it just needs one line to fix I think.

### PR DESCRIPTION
**Pull request by dai-shi**
close #249

### CODE REVIEW

File: src/utils/useAtomCallback.ts [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
 export function useAtomCallback<Result, Arg>(
-  callback: (get: Getter, set: Setter, arg: Arg) => Result
+  callback: (get: Getter, set: Setter, arg: Arg) => Result,
+  scope?: Scope
 ): (arg: Arg) => Promise<Result>
 
 export function useAtomCallback<Result, Arg>(
-  callback: (get: Getter, set: Setter, arg: Arg) => Result
+  callback: (get: Getter, set: Setter, arg: Arg) => Result,
+  scope?: Scope
```
Comment (dai-shi) [inline src/utils/useAtomCallback.ts:18]: This is a bit unfortunate and unexpected, but otherwise this hook is useless with scope.

**Comment by dai-shi** [comment]
Tests would be nice. Is it easy??

**Review by Aslemammad — COMMENTED** [review]
That's great, I'm gonna add the tests, Shall I do that in another PR?

**Comment by dai-shi** [comment]
That would be super helpful. Please directly modify this PR.

**Comment by Aslemammad** [comment]
Sure.

**Review by dai-shi — COMMENTED** [review]
Looks good.

**Comment by dai-shi** [comment]
https://codesandbox.io/s/cocky-sinoussi-p9ssq
Original example reported is fixed too.

**Comment by Aslemammad** [comment]
Great.

**Pull request merged**